application/.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  push:
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run unit tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests.sh
+  api-tests:
+    name: API Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run API tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests-api.sh
+  psr2-check:
+    name: PSR2 check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Check PSR2
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./check-psr2.sh
+
+  build-and-publish:
+    name: Build and Publish
+    needs: [unit-tests,api-tests,psr2-check]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ secrets.DOCKER_USERNAME }}/email-service
```

Comment:
Does this count as exposing the username?

Reply:
No, because "secrets" is part of the encrypted non-public set of configurations in the environment.

Reply:
It would be exposed in the public image that's published. This needs to be changed to something that contains our organization name anyway. I was testing this on my own Docker Hub account.

Reply:
Thanks @briskt. Further questions? Does this need to be a secret? If so, do we want to try to standardize it again, currently `DOCKER_ORG`, but other docker secrets are `DOCKERHUB_*`?

Reply:
Doesn't absolutely need to be a secret, but keeping it secret does enhance security.